testselect model evaluation is failing

```
2026-04-28 23:52:10,432:INFO:bugbug.models.testselect:Testing on 1704 (171 with failures) out of 17032. 3744 schedulable tasks.
  0% 0/1704 [00:00<?, ?it/s]
  0% 0/1704 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/opt/venv/bin/bugbug-train", line 10, in <module>
    sys.exit(main())
             ~~~~^^
  File "/opt/venv/lib/python3.14/site-packages/scripts/trainer.py", line 143, in main
    retriever.go(args)
    ~~~~~~~~~~~~^^^^^^
  File "/opt/venv/lib/python3.14/site-packages/scripts/trainer.py", line 41, in go
    metrics = model_obj.train(limit=args.limit)
  File "/opt/venv/lib/python3.14/site-packages/bugbug/model.py", line 574, in train
    self.evaluation()
    ~~~~~~~~~~~~~~~^^
  File "/opt/venv/lib/python3.14/site-packages/bugbug/models/testselect.py", line 693, in evaluation
    push["all_possibly_selected"] = self.select_tests(
                                    ~~~~~~~~~~~~~~~~~^
        commits, 0.5, push_num - 100
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/venv/lib/python3.14/site-packages/bugbug/models/testselect.py", line 580, in select_tests
    for data in test_scheduling.generate_data(
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self.granularity,
        ^^^^^^^^^^^^^^^^^
    ...<5 lines>...
        tuple(),
        ^^^^^^^^
    ):
    ^
  File "/opt/venv/lib/python3.14/site-packages/bugbug/test_scheduling.py", line 786, in generate_data
    ) = _read_and_update_past_failures(
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        past_failures, "all", runnable, ("all",), push_num, is_regression
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/venv/lib/python3.14/site-packages/bugbug/test_scheduling.py", line 731, in _read_and_update_past_failures
    values_prev_2800.append(value - cur[round((push_num - 2800) / 100)])
                                    ~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.14/site-packages/bugbug/utils.py", line 411, in __getitem__
    assert day >= self.start_day, (
           ^^^^^^^^^^^^^^^^^^^^^
AssertionError: Can't get a day (216) from earlier than start day (217)
```

The reason for this is that the past failure queues cover HISTORICAL_TIMESPAN (4500) pushes. The furthest lookback used when computing features is 2800 pushes (https://github.com/mozilla/bugbug/blob/9157fd518dc9c7c98adfc693179e2130d9622e22/bugbug/test_scheduling.py#L731), leaving only `HISTORICAL_TIMESPAN - 2800 = 1700` pushes of valid evaluation range relative to last_push_num.

--- a/bugbug/models/testselect.py
+++ b/bugbug/models/testselect.py
@@ -690,9 +690,19 @@ class TestSelectModel(Model):
             # past failure data for the push itself.
             # The number 100 comes from the fact that in the past failure data
             # generation we store past failures in batches of 100 pushes.
-            push["all_possibly_selected"] = self.select_tests(
-                commits, 0.5, push_num - 100
+            push_num -= 100
+
+            # Clamp so that the PAST_FAILURES_LOOKBACK_MONTH-push lookback doesn't
+            # fall before the queue's start_day.
+            start_day_max = round(last_push_num / 100) - int(
+                test_scheduling.HISTORICAL_TIMESPAN / 100
             )
+            min_push_num = (
+                start_day_max + int(test_scheduling.PAST_FAILURES_LOOKBACK_MONTH / 100)
+            ) * 100
+            push_num = max(push_num, min_push_num)
+
+            push["all_possibly_selected"] = self.select_tests(commits, 0.5, push_num)
 
         def do_eval(
             executor: concurrent.futures.ProcessPoolExecutor,

--- a/bugbug/test_scheduling.py
+++ b/bugbug/test_scheduling.py
@@ -100,6 +100,12 @@ db.register(
 
 HISTORICAL_TIMESPAN = 4500
 
+# Lookback windows used when computing past failure features. These roughly
+# correspond to one week, two weeks, and one month of pushes on autoland.
+PAST_FAILURES_LOOKBACK_WEEK = 700
+PAST_FAILURES_LOOKBACK_TWO_WEEKS = 1400
+PAST_FAILURES_LOOKBACK_MONTH = 2800
+
 JOBS_TO_CONSIDER = ("test-", "build-")
 JOBS_TO_IGNORE = (
     "docker-image-",
@@ -726,9 +732,15 @@ def _read_and_update_past_failures(
         value = cur[round(push_num / 100)]
 
         values_total.append(value)
-        values_prev_700.append(value - cur[round((push_num - 700) / 100)])
-        values_prev_1400.append(value - cur[round((push_num - 1400) / 100)])
-        values_prev_2800.append(value - cur[round((push_num - 2800) / 100)])
+        values_prev_700.append(
+            value - cur[round((push_num - PAST_FAILURES_LOOKBACK_WEEK) / 100)]
+        )
+        values_prev_1400.append(
+            value - cur[round((push_num - PAST_FAILURES_LOOKBACK_TWO_WEEKS) / 100)]
+        )
+        values_prev_2800.append(
+            value - cur[round((push_num - PAST_FAILURES_LOOKBACK_MONTH) / 100)]
+        )
 
         if is_regression:
             cur[round(push_num / 100)] = value + 1

